### PR TITLE
gobject-introspection: declare variables in .pc file

### PR DESCRIPTION
**Pull request by ericLemanissier**
Specify library name and version:  **gobject-introspection/***

the variables are declared upstream in https://github.com/GNOME/gobject-introspection/blob/master/meson.build#L248

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by SSE4 — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

File: recipes/gobject-introspection/all/test_package/conanfile.py [hunk 1]
```diff
@@ -8,11 +8,12 @@ class TestPackageConan(ConanFile):
 
     def build(self):
         if self.settings.os != 'Windows':
-            self.run('g-ir-annotation-tool --version', run_environment=True)
-            self.run('g-ir-compiler --version', run_environment=True)
-            self.run('g-ir-generate --version', run_environment=True)
-            self.run('g-ir-inspect -h', run_environment=True)
-            self.run('g-ir-scanner --version', run_environment=True)
+            with tools.environment_append({'PKG_CONFIG_PATH': "."}):
```
Comment (madebr) [inline recipes/gobject-introspection/all/test_package/conanfile.py:11]: This should be moved to `test` + only run when not cross building.
It's okay to test for presence of the .pc sections, but it's not okay to run the executables.

Comment (ericLemanissier) [reply recipes/gobject-introspection/all/test_package/conanfile.py:11]: I'm not sure about this, I think these tools are not meant to run on the host platform. They are build tools

Comment (madebr) [reply recipes/gobject-introspection/all/test_package/conanfile.py:11]: When executing the test, won't `g-ir-compiler` be compiled for the host machine and thus potentially not run on the build machine?
Or are these platform independent scripts?

File: recipes/gobject-introspection/all/conanfile.py [hunk 2]
```diff
@@ -110,3 +109,19 @@ def package_info(self):
         bin_path = os.path.join(self.package_folder, "bin")
         self.output.info("Appending PATH env var with: {}".format(bin_path))
         self.env_info.PATH.append(bin_path)
+
+        exe_ext = ".exe" if self.settings.os == "Windows" else ""
+
+        pkgconfig_variables = {
+            'datadir': '${prefix}/res',
+            'bindir': '${prefix}/bin',
+            'g_ir_scanner': '${bindir}/g-ir-scanner',
```
Comment (madebr) [inline recipes/gobject-introspection/all/conanfile.py:118]: What if we're cross building?
Will these executables run when I'm building something for another os/arch?

File: recipes/gobject-introspection/all/conanfile.py [hunk 3]
```diff
@@ -36,21 +37,19 @@ def config_options(self):
             raise ConanInvalidConfiguration("%s recipe does not support windows. Contributions are welcome!" % self.name)
```
Comment (prince-chrismc) [inline recipes/gobject-introspection/all/conanfile.py:37]: Can this be in validate?

**Review by prince-chrismc — APPROVED** [review]
Minor detail not worth blocking

**Pull request merged**